src/test/libradosstriper/TestCase.cc

```diff
@@ -78,3 +81,95 @@ void StriperTestParam::SetUp()
   ASSERT_EQ(0, cluster.ioctx_create(pool_name.c_str(), ioctx));
   ASSERT_EQ(0, RadosStriper::striper_create(ioctx, &striper));
 }
+
+
+std::string MultiStriperTest::pool_name;
+librados::Rados MultiStriperTest::s_cluster;
+
+void MultiStriperTest::init() {
+  connect_cluster_pp(*cluster);
+  cluster->ioctx_create(pool_name.c_str(), *ioctx);
+  RadosStriper::striper_create(*ioctx, striper);
+}
+
+void MultiStriperTest::destory() {
+  ASSERT_EQ(0, destroy_one_pool_pp(pool_name, s_cluster));
+}
+
+void ParallelMultiStriperTest::access(
+  function<int(rados_striper_t&)> rw_access,
+  const size_t n_threads) {
+  refs.clear();
+
+  auto do_access = [&] () {
+    MultiStriperTest *multi_striper = new MultiStriperTest;
+    multi_striper->init();
+    rados_striper_t rsc;
+    RadosStriper::to_rados_striper_t(*(multi_striper->striper), &rsc);
+    multi_striper->ret = rw_access(rsc);
+    std::lock_guard<mutex> guard(m);
+    refs.push_back(multi_striper);
+  };
+
+  vector<thread *> threads;
```

Comment:
could use containers like `list` or `deque` for non-copyable and non-default-constructible types.